### PR TITLE
Upgrade to use AWS SDK V2

### PR DESCRIPTION
**Pull request by sswatosh**
Also upgrades:
 - JDK 8 -> 11
 - Gradle 4 -> 7
 - FindBugs 3 -> SpotBugs 4

There were a variety of existing PMD and SpotBugs warnings present - I tried to fix whatever seemed easy or important, and added excludes for the remaining quibbles. Let me know if any of the excludes feel too aggressive.

### CODE REVIEW

**Review by benjishults — APPROVED** [review]
Love this.  Are there plans to do similar upgrades for other OSS libs?

File: src/main/java/smartthings/ratpack/sqs/DefaultSqsService.java [hunk 1]
```diff
@@ -1,49 +1,52 @@
...
+import software.amazon.awssdk.services.sqs.model.*;
 
 /**
  * Default implementation for communicating with AWS SQS.
  */
 @Singleton
 public class DefaultSqsService implements SqsService, Service {
 
-    private final AmazonSQS sqs;
+    private final SqsClient sqs;
```
Comment (jeff-blaisdell) [inline src/main/java/smartthings/ratpack/sqs/DefaultSqsService.java:18]: Note - the v2 aws lib has an async sqs client (think it's netty based).  It works pretty seemlessly with Ratpack without utilizing blocking threads.  That said - it's a bigger lift then what you have here, and what you have is functionally equivalent.

**Review by jeff-blaisdell — APPROVED** [review]

**Comment by sswatosh** [comment]
> Love this. Are there plans to do similar upgrades for other OSS libs?

Not specifically, but our latest service is trying to stick to V2, so we'll probably upgrade anything else we need (which might not be much).

File: src/main/java/smartthings/ratpack/sqs/DefaultSqsService.java (hunk 1 above)
Comment (sswatosh) [reply src/main/java/smartthings/ratpack/sqs/DefaultSqsService.java:18]: That's a good tip - I'll make a note to come back and give that a look once we can test this version out.

**Pull request merged**